src/API/src/allDataFileBuilder.service.ts

```diff
@@ -0,0 +1,56 @@
+const fs = require('fs');
+import { Inject, Injectable } from '@nestjs/common';
+import { handleLastIngestLock } from 'src/ingest/utils/triggerCsvRebuild';
+import { ExportService } from './export/export.service';
+
+interface LastIngest {
+  ingestion: {
+    ingestTime: string;
+    isLocked: boolean;
+  };
+}
+
+@Injectable()
+export class AllDataFileBuilder {
+  private lastIngestTime: LastIngest;
+  constructor(
+    @Inject(ExportService)
+    private readonly exportService: ExportService,
+  ) {
+    this.lastIngestTime = JSON.parse(
+      fs.readFileSync(process.cwd() + '/../../lastIngest.json', {
```

Comment:
What does this do when `lastIngest.json` doesn't exist? Can you stick in a comment here explaining what would happen?

Reply:
Terrible things. I'll ensure that there is redundancy

Reply:
The handler for creating the file is now called in main